### ISSUE TITLE
[import/order] weird behavior when upgrading to 2.27.0

### ISSUE THREAD

**Issue by ImRodry** [import-js/eslint-plugin-import#2681]
I was on 2.26.0 and was trying to update to 2.27.4 but started getting a weird error in 2 of the files on my project.
Here's my config for the import/order rule:
```json
		"import/order": ["error", {
			"groups": [	"builtin", "external", "internal", "sibling", "index", "parent", "type"],
			"newlines-between": "always",
			"alphabetize": { "order": "asc", "caseInsensitive": true }
		}],
```
Here is what my file currently looks like - this was fine in 2.26.0
![image](https://user-images.githubusercontent.com/38259440/212744446-1ca25ddd-f996-4574-a2a5-c040e97c9ceb.png)
This is what the plugin wants it to look like
![image](https://user-images.githubusercontent.com/38259440/212744486-458bae75-c5a8-49e1-bae0-bd178fae001a.png)
There are many issues with this:

- `"../commands/Utility/poll"` appears right after the builtin modules, before external ones
- `"../config.json"` appears after the `discord.js` external module import but before `mongodb`
- `"../events/crowdinverify"` appears after `mongodb` but before `node-cron`, both external modules

On a different file I have a less obvious but equally weird error:
This is what the file looks like currently
![image](https://user-images.githubusercontent.com/38259440/212744893-fbf2f797-2f0c-457c-a354-8c871a87a080.png)
This is what the plugin wants the file to look like:
![image](https://user-images.githubusercontent.com/38259440/212744956-4064447a-0c43-4df9-986b-4004810a2da5.png)
Here I believe the bug happens because index and config are in the same directory, however, I only want index to show up before sibling files, and config should show up after as it is in the parent directory

I have files with similar structures where this is not an issue, so I'm not sure why it is on these 2 files.

**Comment by ImRodry** [comment]
This may or may not be related to #2669. Since I'm not sure, I decided to make a new issue about it.

**Comment by ljharb** [comment]
I think this was fixed by #2674. Can you try v2.27.5 and see if it works?

**Comment by ImRodry** [comment]
There are still issues with this rule but those existed before 2.27.0 so I'll consider this fixed. Thanks!
